packages/sw-background-sync-queue/src/lib/requestManager.js

```diff
@@ -0,0 +1,136 @@
+import {initializationDefaults} from './constants';
+import queue from './queue.js';
+import responseManager from './responseManager';
+let globalConfig = initializationDefaults;
+let globalCallbacks = {};
+
+/**
+ * Class to handle all the request related
+ * transformations, replaying, event handling
+ * broadcasting back to controlled pages etc.
+ * @class RequestManager
+ */
+class RequestManager {
+	/**
+	 * Initializes the request manager
+	 * stores the callbacks object, maintins global config and
+	 * attaches event handler
+	 * @param {Object} config config that gives values like
+	 * maxAge of requests in iDB
+	 * @param {Object} callbacks object that contains function
+	 * ref for various lifecycle callbacks
+	 *
+	 * @memberOf RequestManager
+	 */
+	async initialize(config, callbacks) {
+		this.attachSyncHandler();
+		globalConfig = Object.assign({}, initializationDefaults, config);
+		globalCallbacks = callbacks;
+		await queue.initialize( globalConfig );
+		await queue.cleanupQueue();
+	}
+
+	/**
+	 * attaches sync handler to replay requests when
+	 * sync event is fired
+	 *
+	 * @memberOf RequestManager
+	 */
+	attachSyncHandler() {
+		self.addEventListener('sync', (event) => {
+			event.waitUntil(this.replayRequests());
+		});
+	}
+
+	/**
+	 * function to start playing requests
+	 * in sequence
+	 * @return {Promise} promise that will resolve whenn all requests complete
+	 *
+	 * @memberOf RequestManager
+	 */
+	replayRequests() {
```

Comment:
Anne's pattern is also pretty easy on the eye to read. I'd hold preference on that pattern over manually tracking the way it's done here too.